drivers/dai/intel/uaol/uaol.c

```diff
@@ -0,0 +1,416 @@
+		if (data_size - i - sizeof(struct ipc4_uaol_tlv) < tlv->length)
+			break;
+
+		switch (tlv->type) {
+		case IPC4_UAOL_IOCTL_TLV_SET_EP_TABLE:
+			if (tlv->length != sizeof(struct ipc4_uaol_set_ep_table))
+				return -EINVAL;
+			ep_table = (struct ipc4_uaol_set_ep_table *)tlv->value;
+			hw_dev = uaol_get_hw_device(ep_table->link_idx);
+			if (!hw_dev)
+				return -EINVAL;
+			ret = pm_device_runtime_get(hw_dev);
+			if (ret)
+				return -EIO;
+			entry.usb_ep_address = (ep_table->entry.usb_ep_number << 1) |
+						ep_table->entry.direction;
+			entry.device_slot_number = ep_table->entry.device_slot_number;
+			entry.split_ep = ep_table->entry.split_ep;
+			ret = uaol_program_ep_table(hw_dev, ep_table->stream_idx, entry, true);
+			if (ret)
+				return -EIO;
+			break;
+		case IPC4_UAOL_IOCTL_TLV_RESET_EP_TABLE:
+			if (tlv->length != sizeof(struct ipc4_uaol_set_ep_table))
+				return -EINVAL;
+			ep_table = (struct ipc4_uaol_set_ep_table *)tlv->value;
+			hw_dev = uaol_get_hw_device(ep_table->link_idx);
+			if (!hw_dev)
+				return -EINVAL;
+			entry.usb_ep_address = (ep_table->entry.usb_ep_number << 1) |
+						ep_table->entry.direction;
+			entry.device_slot_number = ep_table->entry.device_slot_number;
+			entry.split_ep = ep_table->entry.split_ep;
+			ret = uaol_program_ep_table(hw_dev, ep_table->stream_idx, entry, false);
+			if (ret)
+				return -EIO;
+			pm_device_runtime_put(hw_dev);
+			break;
+		case IPC4_UAOL_IOCTL_TLV_SET_EP_INFO:
+			if (tlv->length != sizeof(struct ipc4_uaol_usb_ep_info))
+				return -EINVAL;
+			ep_info = (struct ipc4_uaol_usb_ep_info *)tlv->value;
+			if (ep_info->direction == UAOL_USB_EP_DIRECTION_OUT && ep_info->split_ep)
+				dp->hw_cfg.sio_credit_size =
+					MIN(ep_info->usb_mps, UAOL_MPS_SPLIT_EP);
+			else
+				dp->hw_cfg.sio_credit_size = ep_info->usb_mps;
+			break;
+		default:
+			break;
+		}
+
+		hop = sizeof(struct ipc4_uaol_tlv) + tlv->length;
+		data_ptr += hop;
+	}
+
+	return 0;
+}
+
+static int dai_uaol_probe(const struct device *dev)
+{
+	struct dai_intel_uaol_data *dp = dev->data;
+
+	return pm_device_runtime_get(dp->hw_dev);
+}
+
+static int dai_uaol_remove(const struct device *dev)
+{
+	struct dai_intel_uaol_data *dp = dev->data;
+
+	return pm_device_runtime_put(dp->hw_dev);
+}
+
+static int dai_uaol_config_set(const struct device *dev, const struct dai_config *cfg,
+			       const void *bespoke_cfg)
+{
+	struct dai_intel_uaol_data *dp = dev->data;
+	int ret;
+
+	if (!bespoke_cfg) {
+		return -EINVAL;
+	}
+
+	if (!cfg) {
+		return dai_uaol_parse_ioctl_data(dev, bespoke_cfg);
+	}
+
+	ret = dai_uaol_parse_aux_data(dev, bespoke_cfg);
+	if (ret) {
+		return ret;
+	}
+
+	dp->hw_cfg.channels = cfg->channels;
+	dp->hw_cfg.sample_rate = cfg->rate;
+	dp->hw_cfg.sample_bits = cfg->word_size;
+	dp->hw_cfg.channel_map = cfg->link_config;
+
+	dp->dai_state = DAI_STATE_PRE_RUNNING;
+
+	return 0;
+}
+
+static int dai_uaol_config_get(const struct device *dev, struct dai_config *cfg, enum dai_dir dir)
+{
```

Comment:
`dir` is an unused argument. To prevent unnecessary compiler warnings and clearly indicate that this parameter is intentionally unused, please apply the `ARG_UNUSED` macro.

Reply:
Fixed. Thanks !